### PR TITLE
[Internal] Query: Fixes issue with distributed query GetItemQueryStreamIterator

### PR DESCRIPTION
**Pull request by sc978345**
## Description

This change fixes an issue with GetItemQueryStreamIterator for distributed query where containerRid was not getting set in the responseHeaders, leading to an exception when building the response.

We must set the container resource id in the CosmosQueryContext, prior to building the DistributedQueryPipelineStage.

## Type of change

- [] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/DistributedQueryClientTests.cs [hunk 1]
```diff
@@ -260,7 +288,55 @@ private static async Task RunTestsAsync(Container container, IEnumerable<TestCas
...
+                        EnableDistributedQueryGatewayMode = true,
+                    };
+
+                    List<int> extractedResults = new List<int>();
+                    await foreach (ResponseMessage response in RunSimpleQueryAsync(
+                        container,
+                        testCase.Query,
+                        options))
+                    {
+                        Assert.IsTrue(response.StatusCode == System.Net.HttpStatusCode.OK);
```
Comment (neildsh) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/DistributedQueryClientTests.cs:314]: >IsTrue [](http://example.com/codeflow?start=31&length=6)

[Nit] Assert.AreEqual

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/DistributedQueryClientTests.cs [hunk 2]
```diff
@@ -260,7 +288,55 @@ private static async Task RunTestsAsync(Container container, IEnumerable<TestCas
...
+
+                        using (MemoryStream memstr = new MemoryStream())
```
Comment (neildsh) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/DistributedQueryClientTests.cs:316]: >memstr [](http://example.com/codeflow?start=44&length=6)

[Coding Convention] Please do not use abbreviations. consider renaming to `memoryStream` or `stream` instead.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/DistributedQueryClientTests.cs [hunk 3]
```diff
@@ -2,9 +2,12 @@ namespace Microsoft.Azure.Cosmos.EmulatorTests.Query
 {
     using System;
     using System.Collections.Generic;
+    using System.IO;
     using System.Linq;
     using System.Threading.Tasks;
+    using System.Xml.Linq;
```
Comment (neildsh) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/DistributedQueryClientTests.cs:8]: >using System.Xml.Linq; [](http://example.com/codeflow?start=4&length=22)

Please check if this is required

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/DistributedQueryClientTests.cs [hunk 4]
```diff
@@ -201,7 +229,7 @@ private static async Task ContinuationTestsAsync(Container container, IEnumerabl
                 foreach (int pageSize in testCase.PageSizes)
                 {
                     List<int> results = await RunContinuationBasedQueryTestAsync(container, testCase.Query, pageSize);
-                    testCase.ValidateResult(results);
+                    Assert.IsTrue(testCase.ValidateResult(results));
```
Comment (neildsh) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/DistributedQueryClientTests.cs:232]: >Assert.IsTrue [](http://example.com/codeflow?start=20&length=13)

good catch!

**Review by neildsh — APPROVED** [review]

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/DistributedQueryClientTests.cs (hunk 3 above)
Comment (sc978345) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/DistributedQueryClientTests.cs]: It is not -- removed!

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/DistributedQueryClientTests.cs (hunk 2 above)
Comment (sc978345) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/DistributedQueryClientTests.cs]: Renamed

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/DistributedQueryClientTests.cs (hunk 1 above)
Comment (sc978345) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/DistributedQueryClientTests.cs]: Changed

**Review by sboshra — APPROVED** [review]
:shipit:

**Review by neildsh — APPROVED** [review]

**Pull request merged**